pow/src/test/java/tech/pegasys/teku/pow/Web3jEth1MonitorableProviderTest.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.ExecutionException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthChainId;
+import org.web3j.protocol.core.methods.response.EthSyncing;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.async.StubAsyncRunner;
+import tech.pegasys.teku.infrastructure.time.StubTimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.util.config.Constants;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class Web3jEth1MonitorableProviderTest {
+  final Web3j web3 = mock(Web3j.class);
+  final Request request1 = mock(Request.class);
+  final Request request2 = mock(Request.class);
+  StubAsyncRunner asyncRunner;
+  StubTimeProvider timeProvider;
+  Web3jEth1Provider provider;
+
+  static final String CHAIN_ID_CORRECT = "5";
+  static final String CHAIN_ID_WRONG = "1";
+
+  @BeforeEach
+  void setup() {
+    asyncRunner = new StubAsyncRunner();
+    timeProvider = StubTimeProvider.withTimeInSeconds(1000);
+    provider = new Web3jEth1Provider("0 test.host.org", web3, asyncRunner, timeProvider);
+  }
+
+  @Test
+  void shouldUpdateCallTimestampOnSuccess() throws ExecutionException, InterruptedException {
+    succeedACall();
+
+    assertThat(provider.getLastCallTime()).isEqualTo(UInt64.valueOf(1000));
+    assertThat(provider.getLastValidationTime()).isNotEqualTo(UInt64.valueOf(1000));
+  }
+
+  @Test
+  void shouldUpdateCallTimestampOnFailure() {
+    failACall();
+
+    assertThat(provider.getLastCallTime()).isEqualTo(UInt64.valueOf(1000));
+    assertThat(provider.getLastValidationTime()).isNotEqualTo(UInt64.valueOf(1000));
+  }
+
+  @Test
+  void shouldUpdateValidationTimestamp() {
+    prepareRequestWithChainidResponse(request1, CHAIN_ID_WRONG);
+    when(web3.ethChainId()).thenReturn(request1);
+    prepareRequestWithSyncingResponse(request2, false);
+    when(web3.ethSyncing()).thenReturn(request2);
+
+    try {
+      provider.validate().get();
+    } catch (Exception ignored) {
+    }
```

Comment:
Probably safer and simpler to just use:
```suggestion
    assertThat(provider.validate()).isCompleted();
```

Since we know the future should complete immediately (web3 is a mock and StubAsyncRunner doesn't actually use other threads).